lib/icinga/clusterevents.cpp

```diff
@@ -619,7 +619,29 @@ Value ClusterEvents::AcknowledgementClearedAPIHandler(const MessageOrigin::Ptr&
 
 Value ClusterEvents::ExecuteCommandAPIHandler(const MessageOrigin::Ptr& origin, const Dictionary::Ptr& params)
 {
-	EnqueueCheck(origin, params);
+	ApiListener::Ptr listener = ApiListener::GetInstance();
+	if (!listener)
+		return Empty;
+
+	Endpoint::Ptr execEndpoint = Endpoint::GetByName(params->Get("endpoint"));
+	if (execEndpoint != Endpoint::GetLocalEndpoint()) {
+
+		Zone::Ptr endpointZone = execEndpoint->GetZone();
+		Zone::Ptr localZone = Zone::GetLocalZone();
+
+		if (!endpointZone->IsChildOf(localZone)) {
+			return Empty;
+		}
+
+		Dictionary::Ptr execMessage = new Dictionary();
+		execMessage->Set("jsonrpc", "2.0");
+		execMessage->Set("method", "event::ExecuteCommand");
+		execMessage->Set("params", params);
+
+		listener->RelayMessage(origin, endpointZone, execMessage, true);
+	} else {
+		EnqueueCheck(origin, params);
+	}
```

Comment:
... instead of all this stuff...

Reply:
Probably putting the `endpointZone` in the RelayMessage we can remove this if
```
if (!endpointZone->IsChildOf(localZone)) {
	return Empty;
}
```
or not?